scene/gui/tree.cpp

```diff
@@ -682,7 +704,21 @@ TreeItem *TreeItem::get_prev_visible(bool p_wrap) {
 	return current;
 }
 
-TreeItem *TreeItem::get_next_visible(bool p_wrap) {
+TreeItem *TreeItem::get_prev_visible(bool p_wrap) {
+	TreeItem *loop = this;
+	TreeItem *prev = this->_get_prev_visible(p_wrap);
+	while (prev && !prev->is_visible()) {
+		prev = prev->_get_prev_visible(p_wrap);
+		if (prev == loop) {
+			// Check that we haven't looped all the way around to the start.
+			prev = nullptr;
+			break;
+		}
+	}
+	return prev;
+}
```

Comment:
I think it's better to integrate it into the original method

Reply:
I believe I have fixed the other issues, but I am not sure how I would integrate this into the original method. If it were just called once then it would be easy, but because we need to call it twice in a recursive fashion I am not sure if this is possible? Or maybe I am just blind and/or dumb 😅